examples/stable_diffusion_v2/ldm/modules/attention.py

```diff
@@ -142,15 +144,19 @@ def __init__(
         )
 
         self.attention = Attention(dim_head, upcast=upcast)
+        self.fa_max_head_dim = fa_max_head_dim
 
         self.enable_flash_attention = (
             enable_flash_attention and FLASH_IS_AVAILABLE and (ms.context.get_context("device_target") == "Ascend")
         )
         if self.enable_flash_attention:
-            self.flash_attention = FlashAttention(
-                head_dim=dim_head, head_num=heads, high_precision=True
-            )  # TODO: how high_precision affect the training or inference quality
-            self.fa_mask_dtype = ms.uint8  # choose_flash_attention_dtype()
+            # TODO: how high_precision affect the training or inference quality
+            if version.parse(ms.__version__) < version.parse("2.2"):
+                self.flash_attention = FlashAttention(head_dim=dim_head, high_precision=True)
+                self.fa_mask_dtype = ms.float16  # choose_flash_attention_dtype()
```

Comment:
it [looks like](https://gitee.com/mindspore/mindformers/blob/dev/mindformers/version_control.py#L194) MS2.2.0 requires fp16:
`if version.parse(ms.__version__) <= version.parse("2.2"):`